save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/tables/TableComponent.kt

```diff
@@ -197,52 +224,46 @@ fun <D : Any, P : TableProps<D>> tableComponent(
                                         this.className = className
-                                        +column.render("Header")
-                                        // fixme: find a way to set `canSort`; now it's always true
-                                        if (column.canSort) {
-                                            spread(columnProps)
+                                        child(
+                                            renderHeader(header)
+                                        )
+                                        if (column.getCanSort()) {
+                                            style = style ?: jso {}
+                                            style?.cursor = "pointer".unsafeCast<Cursor>()
                                             span {
-                                                +when {
-                                                    column.isSorted -> " 🔽"
-                                                    column.isSortedDesc -> " 🔼"
+                                                +when (column.getIsSorted()) {
+                                                    SortDirection.asc -> " 🔽"
+                                                    SortDirection.desc -> " 🔼"
                                                     else -> ""
                                                 }
                                             }
+                                            onClick = column.getToggleSortingHandler()
                                         }
                                     }
                                 }
                             }
                         }
                     }
                     tbody {
-                        spread(tableInstance.getTableBodyProps())
-                        tableInstance.page.map { row ->
-                            tableInstance.prepareRow(row)
+                        tableInstance.getRowModel().rows.map { row ->
                             tr {
-                                spread(row.getRowProps(getRowProps(row)))
-                                row.cells.map { cell ->
-                                    // fixme: userProps are not present in actual html, but .render("Cell") produces td, so can't wrap
-                                    child(cell.render("Cell", userProps = json().apply {
-                                        spread(cell.getCellProps()) { key, value ->
-                                            this[key] = value
-                                        }
-                                    }))
+                                spread(getRowProps(row))
+                                row.getVisibleCells().map { cell ->
+                                    child(renderCell(cell))
                                 }
                             }
-                            if (row.isExpanded) {
+                            if (row.getIsExpanded()) {
```

Comment:
maybe add extension in `TableUtil` to `get`?

Reply:
Added getters for `isExpanded` and `can(Next/Previous)Page`